crates/subspace-networking/src/node_runner.rs

```diff
@@ -348,6 +348,16 @@ where
                 num_established,
                 ..
             } => {
+                // Save known addresses that were successfully dialed.
+                if let ConnectedPoint::Dialer { address, .. } = &endpoint {
+                    // filter non-global addresses when non-globals addresses are disabled
+                    if is_global_address_or_dns(address) || self.allow_non_global_addresses_in_dht {
```

Comment:
Strictly speaking `self.allow_non_global_addresses_in_dht` is much cheaper operation than `is_global_address_or_dns(address)` and should go first